shimmingtoolbox/load_nifti.py

```diff
@@ -0,0 +1,165 @@
+#!usr/bin/env python3
+# -*- coding: utf-8
+
+import os
+import logging
+import numpy as np
+import nibabel as nib
+import json
+import math
+
+logger = logging.getLogger(__name__)
+PHASE_SCALING_SIEMENS = 4096
+
+
+def load_nifti(path_data):
+    """
+    Load data from a NIFTI type file with dcm2bids.
+    Args:
+        path_data (str): absolute or relative path to the directory the acquisition data
+    Returns:
+        info (Nifti1Image.Header): List containing all information from every Nifti image
+        json_info (dict): List containing all information in JSON format from every Nifti image
+        niftis (ndarray): 5D array of all acquisition in time (x, y, z, echo, volume)
+
+    Note:
+        If 'path' is a folder containing niftis, directly output niftis. It 'path' is a folder containing acquisitions,
+        ask the user for which acquisition to use.
+    """
+    if not os.path.exists(path_data):
+        raise RuntimeError("Not an existing NIFTI path")
+
+    file_list = []
+    [file_list.append(os.path.join(path_data, f)) for f in os.listdir(path_data) if f not in file_list]
+
+    nifti_path = ""
+    if all([os.path.isdir(f) for f in file_list]):
+        acquisitions = [f for f in file_list if os.path.isdir(f)]
+        logging.info("Multiple acquisition directories in path. Choosing only one.")
+    elif all([os.path.isfile(f) for f in file_list]):
+        logging.info("Acqusition directory given. Using acquisitions.")
+        nifti_path = path_data
+    else:
+        raise RuntimeError("Directories and files in input path")
+
+    if not nifti_path:
+        for i in range(len(acquisitions)):
+            logging.info("{}:{}\n".format(i, os.path.basename(file_list[i])))
+            print("{}:{}\n".format(i, os.path.basename(file_list[i])))
+
+        select_acquisition = -1
+        while 1:
+            input_resp = input("Enter the number for the appropriate acquisition folder, (type 'q' to quit) : ")
+            if input_resp == 'q':
+                return 0
+
+            select_acquisition = int(input_resp)
+
+            if select_acquisition in range(len(acquisitions)):
+                break
+            else:
+                logging.error("Input must be linked to an acquisition folder. {} is out of range".format(input_resp))
+
+        nifti_path = os.path.abspath(file_list[select_acquisition])
+
+    nifti_list = [os.path.join(nifti_path, f) for f in os.listdir(nifti_path) if f.endswith((".nii", ".nii.gz"))]
+    n_echos = len(nifti_list)
```

Comment:
This correctly implements the matlab function but it should be revisited as the number of files is not always the number of echoes. What if there are multiple runs, multiple time points, etc. 

The header should probably be read to figure out the appropriate way to "stitch" the files together.

Reply:
True. I should add the ability to detect the correct way to parse through data. From what I've seen for multiple runs, the files are in separate folders which you can choose before hand. But, what do you mean by time points?

Reply:
If you mean that we should parse the description of the header file, is it really reliable as information?

Reply:
@po09i  I think that the NIFTI files generated by dcm2bids will be always as different echoes for each file and the files will contain the rest of the data. Tell me if I'm wrong. I have not seen another example on the BIDS website.

Reply:
I found this link to the [bids convention](https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/01-magnetic-resonance-imaging-data.html#the-run-entity) really useful. It describes what runs are and other specific bids format information. 

We used to separate scans in separate acquisitions however now that we are trying to deal with only having one config file, it is not a guaranty that one acquisition will be in the folder (If we run 2 fieldmaps for example). I think counting the number of files in a folder is not a rigorous check and the only way to make sure is indeed reading the header info.

Reply:
let's not spend too much time on this, as it is still unclear how the typical experimental workflow will look like (e.g. will we impose convention for naming runs during acquisition? will we only select the lastest fieldmap? etc.). For now, let's keep it simple and as experimental scenario are clearer (meaning: post-covid19: we can scan again) we will revisit this

Reply:
I changed the function so it will detect if there are multiple runs in the same folder and ask for which one to load. It should be clarified for later on.